src/utils/wrappers.py

```diff
@@ -0,0 +1,343 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import ee
+
+
+class CollectionClass:
+    """Base Imagery class for ImageCollection."""
+
+    def __init__(self, link, bands=None):
+        """Initialize the Collection class.
+
+        Parameters
+        ----------
+        link : str
+            ImageCollection link
+        bands : list[str], optional
+            Name of bands to select. The default is None to take all bands.
+
+        """
+        self.bands = bands
+        self.imagery = ee.ImageCollection(link)
+        if self.bands:
+            self.imagery = self.imagery.select(self.bands)
+
+    def get_clipped_bands(self, geometry):
+        """Get clipped images after a filter bound on geometry.
+
+        Parameters
+        ----------
+        geometry : ee.Geometry
+            Geometry to clip to
+
+        Returns
+        -------
+        ee.ImageCollection
+            cliped images filterbounded on geometry
+
+        """
+        images = self.imagery.filterBounds(geometry)
+        return images.map(lambda x: x.clip(geometry))
+
+    def get_clipped_range_bands(self, geometry, start_date, end_date):
+        """Get clipped images from a specific date range.
+
+        Parameters
+        ----------
+        geometry : ee.Geometry
+            DESCRIPTION.
+        start_date : String/EE.Date
+            DESCRIPTION.
+        end_date : String/EE.Date
+            DESCRIPTION.
+
+        Returns
+        -------
+        TYPE
+            DESCRIPTION.
+
+        """
+        images = self.imagery.filterBounds(geometry)
+
+        def clip_func(image):
+            return image.clip(geometry)
+        return images.filterDate(start_date, end_date).map(clip_func)
+
+
+class RoadImagery:
+    """Road Imagery Class: Wrapper for ee.Image."""
+
+    def __init__(self, link, bands):
+        """Initialize the Imagery Class.
+
+        Parameters
+        ----------
+        link : str
+            ee.Image link
+        bands : list[str], optional
+            Name of bands to select. The default is None to take all bands.
+
+        Returns
+        -------
+        None.
+
+        """
+        self.bands = bands
+        self.imagery = ee.Image(link).select(self.bands)
+
+    def get_bands(self, geometry):
+        """.
+
+        Parameters
+        ----------
+        geometry : ee.Geometry
+            Geometry to clip to
+
+
+        Returns
+        -------
+        ee.Image
+            Image clipped to the geometry
+
+        """
+        return self.imagery.clip(geometry).unmask(0, False)
+
+
+class TropomiImagery(CollectionClass):
+    """."""
+
+    def __init__(self, link, bands=None, before_range=(0, 'day'),
+                 after_range=(7, 'day')):
+        """.
+
+        Parameters
+        ----------
+        link : TYPE
+            DESCRIPTION.
```

Comment:
Did you mean to fill out these descriptions?

Reply:
Done!